src/libarena/lib.rs

```diff
@@ -106,7 +106,7 @@ impl<T> TypedArenaChunk<T> {
         unsafe {
             if mem::size_of::<T>() == 0 {
                 // A pointer as large as possible for zero-sized elements.
-                !0 as *mut T
+                (!0usize) as *mut T
```

Comment:
I don't understand why the original code did the correct thing. `!0` on its own is an i32 (value -1). i32 cast to a pointer is valid, yet this seems to infer using usize.